### DISCUSSION TITLE
Adding a `dry-run` sub command to diesel-migration

### DISCUSSION THREAD

**Discussion by RitickMadaan** [diesel-rs/diesel#3825] [Ideas]
I would like to add a dry-run sub command to diesel-migration. Running `diesel migration dry-run` will print to stdout all the sql queries which will be run by `diesel migration run`. This will give more visibility and be helpful with using diesel in production environments where I would preferably like to see at a single place, all the queries I am going to run.

**Comment by weiznich** [comment]
I think that's an good idea, although I would likely use `diesel migration run --dry-run` for this instead to be more close on what for example cargo does for `cargo publish`. We are happy to mentor work on this feature and to receive a PR for this feature.

**Reply by RitickMadaan** [reply]
>  although I would likely use `diesel migration run --dry-run` for this instead to be more close on what for example cargo does for cargo publish

sure, this sounds good @weiznich. I'll raise a PR : )
